src/status_im/wallet_connect_legacy/core.cljs

```diff
@@ -0,0 +1,188 @@
+  [{:keys [db]} session account]
+  (let [connector (:connector session)
+        address (:address account)]
+    (^js .updateSession connector (clj->js {:chainId 1 :accounts [address]}))
+    {:hide-wallet-connect-app-management-sheet nil
+     :db (assoc db :wallet-connect/showing-app-management-sheet? false)}))
+
+(fx/defn disconnect-session
+  {:events [:wallet-connect-legacy/disconnect]}
+  [{:keys [db]} session]
+  (let [sessions (get db :wallet-connect-legacy/sessions)
+        connector (:connector session)]
+    (^js .killSession connector)
+    {:hide-wallet-connect-app-management-sheet nil
+     :hide-wallet-connect-success-sheet nil
+     :db (-> db
+             (assoc :wallet-connect-legacy/sessions (remove session sessions))
+             (dissoc :wallet-connect/session-managed))}))
+
+(fx/defn pair-session
+  {:events [:wallet-connect-legacy/pair]}
+  [{:keys [db]} {:keys [data]}]
+  (let [connector (wallet-connect-legacy/create-connector data)
+        wallet-connect-enabled? (get db :wallet-connect/enabled?)]
+    (when wallet-connect-enabled?
+      (do
+        (^js .on connector "session_request" (fn [error payload]
+                                               (re-frame/dispatch [:wallet-connect-legacy/proposal payload connector])))
+        (^js .on connector "connect" (fn [error payload]
+                                       (re-frame/dispatch [:wallet-connect-legacy/created payload])))
+        (^js .on connector "call_request" (fn [error payload]
+                                            (re-frame/dispatch [:wallet-connect-legacy/request-received (js->clj payload :keywordize-keys true) connector])))
+        (^js .on connector "session_update" (fn [error payload]
+                                              (re-frame/dispatch [:wallet-connect-legacy/update-sessions (js->clj payload :keywordize-keys true) connector])))))
+    (merge
+     {:dispatch [:navigate-back]}
+     (when wallet-connect-enabled?
+       {:db (assoc db :wallet-connect-legacy/scanned-uri data)}))))
+
+(fx/defn update-sessions
+  {:events [:wallet-connect-legacy/update-sessions]}
+  [{:keys [db] :as cofx} payload connector]
+  (let [sessions (get db :wallet-connect-legacy/sessions)
+        accounts-new (:accounts (first (:params payload)))
+        session (first (filter #(= (:connector %) connector) sessions))
+        updated-session (assoc-in session [:params 0 :accounts] accounts-new)]
+    (println session updated-session (.indexOf sessions session) "LELEL")
+    {:db (-> db
+             (assoc :wallet-connect-legacy/sessions (assoc sessions (.indexOf sessions session) updated-session))
+             (dissoc :wallet-connect/session-managed))}))
+
+(fx/defn wallet-connect-legacy-complete-transaction
+  {:events [:wallet-connect-legacy.dapp/transaction-on-result]}
+  [{:keys [db]} message-id connector result]
+  (let [response {:id message-id
+                  :result result}]
+    (.approveRequest connector (clj->js response))
```

Comment:
effects

Reply:
@briansztamfater is there any reason why this one was not moved to fx?

Reply:
Already moved to fx but forgot to remove this line, removing it now